tests/end2end/verify.py

```diff
@@ -0,0 +1,47 @@
+"""Tool to verify that checkpoints match."""
+
+import argparse
+import torch
+import numpy as np
+
+parser = argparse.ArgumentParser(description="Compare checkpoints for testing.")
+parser.add_argument("--new-model", help="The path to the new model.")
+parser.add_argument(
+    "--old-model", help="The path to the model we are comparing aganist."
+)
+parser.add_argument("--compare")
+
+
+def get_compare_function(compare):
+    """Eventually add configurable comaprison functions?"""
+
+    def _cmp(a, b):
+        return np.allclose(a.numpy(), b.numpy())
```

Comment:
Is allclose appropriate for all tests here? I'd think if we were checking e.g. that the checked-out version matches the original version, we can enforce equality, right? Since the data should be byte-for-byte equivalent using tensorstore?